src/components/GithubAuth.jsx

```diff
@@ -0,0 +1,36 @@
+import { auth, githubProvider } from "../utils/firebaseConfig";
+import { signInWithPopup } from "firebase/auth";
+import { toast } from "react-toastify";
+import { FaGithub } from "react-icons/fa";
+import { Button } from "./ui/button";
+import { useSharedState } from "@/context/sharedContext";
+
+const GithubAuth = () => {
+  const { dispatch } = useSharedState();
+  const notifyError = (message) => toast.error(message);
+
+  const signInWithGithub = async () => {
+    try {
+      await signInWithPopup(auth, githubProvider);
+      if (auth.currentUser) {
+        toast.success("Logged in successfully");
+        dispatch({ type: "HIDE_MODAL" });
```

Comment:
Change this event. Almost every web app has more than one modal, so you want to be specific with your naming.